### PR TITLE
Throw a lint error or an exception when no player owns the world

### PR DESCRIPTION
**Pull request by abcdefg30**
Requested in https://github.com/OpenRA/OpenRAModTemplate/pull/3#discussion_r115697809.

### CODE REVIEW

File: OpenRA.Mods.Common/Lint/CheckPlayers.cs [hunk 1]
```diff
@@ -34,10 +35,20 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
...
+					if (player.Enemies.Any() || player.Allies.Any())
+						emitWarning("The player {0} owning the world should not have any allies or enemies.".F(player.Name));
+
+					if (player.Playable)
+						emitError("The player {0} owning the world can't be playable.".F(player.Name));
+				}
 			}
 
+			if (!worldOwnerFound)
+				emitError("No player which owns the world found.");
```
Comment (rob-v) [inline OpenRA.Mods.Common/Lint/CheckPlayers.cs]: minor nit: could be 'No player owning the world...' like above

File: OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs [hunk 2]
```diff
@@ -25,16 +26,24 @@ public void CreatePlayers(World w)
 		{
 			var players = new MapPlayers(w.Map.PlayerDefinitions).Players;
 			var worldPlayers = new List<Player>();
+			var ownerAdded = false;
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: minor nit: worldOwnerFound vs ownerAdded

File: OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs [hunk 3]
```diff
@@ -25,16 +26,24 @@ public void CreatePlayers(World w)
...
+
 				if (kv.Value.OwnsWorld)
+				{
+					ownerAdded = true;
 					w.SetWorldOwner(player);
+				}
 			}
 
+			if (!ownerAdded)
+				throw new InvalidOperationException("Map {0} does not define a player actor owning the world.".F(w.Map.Title));
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: this isn't good place to throw exception - the game now crashes in load screen. before game normally started.

Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: Where should it crash ideally?

Comment (rob-v) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: I don't know if it has to crash, didn't before. Ideally there would be error message (e.g. on map selection, before start,..) to inform player, crashing to desktop while loading the game looks bad.

Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: Well, it didn't crash before, but @pchote explicitly requested a crash in https://github.com/OpenRA/OpenRAModTemplate/pull/3#discussion_r115697809.

**Comment by abcdefg30** [comment]
Updated.

**Comment by rob-v** [comment]
👍

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1c3b36d4916eb90cbb62f27d5fb76df953036ed6...1931bc8d78ec3d31061d05fb4e68edb46fce8704

**Pull request merged**